enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,381 @@
+  - @sdodson - as the staff engineer most closely tied to install experience
+creation-date: 2021-05-04
+last-updated: 2021-05-04
+status: provisional
+---
+
+# User Selectable Install Solutions
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposes a mechanism for cluster installers to exclude one or more optional components for
+their installation which will determine which payload components are/are not installed in their cluster.
+Core components are defined as the set of Second Level Operators managed by the Cluster Version Operator
+which today cannot be disabled until after completing the install and editing a CVO override, or editing
+the CVO overrides as part of rendering+editing manifests.  In addition, using CVO overrides put the cluster
+into an unsupported and un-upgradeable state, making it insufficient as a solution.
+
+The proposed UX is to make this a first class part of the install config api with the implementation
+being arguments supplied to the CVO to filter out the user-selected manifests.
+
+## Motivation
+
+There is an increasing desire to move away from "one size fits all" cluster installations, and
+towards flexibility about what should/should not exist in a new cluster out of the box.  This can
+be seen in efforts such as hypershift, single node, and code-ready-containers.  Each of these
+efforts has done some amount of one-off work to enable their requirements.  This EP proposes a
+mechanism that allows components to be disabled in a first class way that the installer exposes.
+
+### Goals
+
+* Admins can easily explicitly exclude specific "optional" components from their cluster, at install time.
+* Admins can enable a previously excluded optional component, at runtime.
+* Install wrappers like assisted-installer can define an install-config that excludes specific components
+* Define an api that could be used in the future to exclude cluster capabilities based on things other than
+CVO filtering, such as turning off a particular api for the cluster.
+
+### Non-Goals
+
+* Making control-plane critical components optional (k8s apiserver, openshift apiserver, openshift controller,
+  networking, etc)
+* Defining which components should be disable-able (this will be up to component teams to classify themselves
+as `capabilities` or not)
+* Providing a way to install OLM operators as part of the initial cluster install.  This EP is about making
+the install experience around the existing CVO-based components more flexible, not adding new components to the
+install experience.
```

Comment:
I see some [mentions in threads][1] about moving components to OLM, and that's certainly my current preference.  This "no day-1 OLM" entry is currently phrased as "we're not talking about additional components", and that's fine, but it means that "move the console to OLM and out of the core, and folks who want a console can install the OLM console" isn't directly addressed today.  Obviously some install-time things (networking, etc.) are hard to move out of OLM, because they're important at bootstrapping.  But neither console nor samples are involved in bootstrapping, and you'd only need something that worked post-bootstrap-removal to install those via OLM.  I don't know enough about OLM to say how difficult setting those up via manifests would be, but my naive impression is that it would be possible.  Even if we end up rejecting "move the optional stuff into OLM", I think it's worth a subsection under *Alternatives* laying out the pros and cons in more detail to help make an informed opinion.

[1]: https://github.com/openshift/enhancements/pull/922#discussion_r731190381